packages/block-library/src/query-total/edit.js

```diff
@@ -0,0 +1,98 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	useBlockProps,
+	InspectorControls,
+	BlockControls,
+} from '@wordpress/block-editor';
+import {
+	PanelBody,
+	ToolbarGroup,
+	ToolbarDropdownMenu,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { resultsFound, displayingResults } from './icons';
+
+export default function QueryTotalEdit( { attributes, setAttributes } ) {
+	const { displayType } = attributes;
+
+	// Block properties and classes.
+	const blockProps = useBlockProps();
+
+	const getButtonPositionIcon = () => {
+		switch ( displayType ) {
+			case 'total-results':
+				return resultsFound;
+			case 'range-display':
+				return displayingResults;
+		}
+	};
+
+	const buttonPositionControls = [
+		{
+			role: 'menuitemradio',
+			title: __( 'Total results' ),
+			isActive: displayType === 'total-results',
+			icon: resultsFound,
+			onClick: () => {
+				setAttributes( { displayType: 'total-results' } );
+			},
+		},
+		{
+			role: 'menuitemradio',
+			title: __( 'Range display' ),
+			isActive: displayType === 'range-display',
+			icon: displayingResults,
+			onClick: () => {
+				setAttributes( { displayType: 'range-display' } );
+			},
+		},
+	];
+
+	// Controls for the block.
+	const controls = (
+		<>
+			<BlockControls>
+				<ToolbarGroup>
+					<ToolbarDropdownMenu
+						icon={ getButtonPositionIcon() }
+						label={ __( 'Change display type' ) }
+						controls={ buttonPositionControls }
+					/>
+				</ToolbarGroup>
+			</BlockControls>
+			<InspectorControls>
+				<PanelBody title={ __( 'Display Settings' ) }>
+					<p>
+						{ __( 'Choose the type of information to display.' ) }
+					</p>
+				</PanelBody>
+			</InspectorControls>
+		</>
+	);
+
+	// Render output based on the selected display type.
+	const renderDisplay = () => {
+		if ( displayType === 'total-results' ) {
+			return <div>{ __( '12 results found' ) }</div>;
```

Comment:
My last question is where the number 12 comes from. I'm not sure how much these display elements need to represent real data, but in some cases, it would be possible to show the actual number of results and the current settings for items per page.

Reply:
@gziolo We have a precedence for this in core. The query pagination also doesn't show the real pagination but has a good default value that showcases the block in action 🤔

I think it makes sense here also as `1 - 1 out of 1` doesn't really make sense

Reply:
> I think it makes sense here also as 1 - 1 out of 1 doesn't really make sense

Now that you shared it. What should happen for edge cases?

- `1 -1 out of 1` - should it simply display `1` or an empty string?
- `1-5 out of 5` - should it display `1-5`?

Thank you for the explanation about the number `12`. It's now documented and we can proceed further 😄

Reply:
I think the proper answer would still be: 

1 - 1 out of 1

The format should be:

`index of first item on this page` - `index of last item on this page` out of `total number of results`

Reply:
Currently this is how it is handled on the frontend
<img width="269" alt="Screenshot 2024-12-10 at 1 37 00 PM" src="https://github.com/user-attachments/assets/0bc30316-f31c-40d1-a700-ce51fcc48ba3">
<img width="195" alt="Screenshot 2024-12-10 at 1 37 07 PM" src="https://github.com/user-attachments/assets/6d03de1a-94b5-434c-9552-a54ac7b8248f">

Reply:
Looking at the logic, it will show `Displaying 1 of 1` which is perfectly fine.